extensions/amp-animation/0.1/amp-animation.js

```diff
@@ -42,17 +60,125 @@ export class AmpAnimation extends AMP.BaseElement {
     user().assert(isExperimentOn(this.win, TAG),
         `Experiment "${TAG}" is disabled.`);
 
+    // Trigger.
+    const trigger = this.element.getAttribute('trigger');
+    if (trigger) {
+      this.triggerOnVisibility_ = user().assert(
+          trigger == 'visibility',
```

Comment:
So, it seems for embeds it is driven by the container `viewportCallback`, but more non-embedded it is just doc visibility?